File: java8/Fib.java
import java.util.HashMap;
import java.util.Map;

public class Fib {
  private final static Map<Integer,Long> cache = new HashMap<Integer, Long>() {{
    put(0,0L);
    put(1,1L);
  }};
  public static long fib(int x) {
    return cache.computeIfAbsent(x, n -> fib(n-1) + fib(n-2));
  }

  public static void main(String[] args) {
    System.out.println(fib(10));
  }
}

File: fib.go
package main

import (
    "fmt"
)

func main() {
    fmt.Printf("%d\n", fib(15))
}

func fib(n int) int {
    x, y := 0, 1
    for i := 0; i<n; i++ {
        x, y = y, x+y
    }
    return x
}
